### ISSUE TITLE
request for nvim commands to remap as we see fit. 

### ISSUE THREAD

**Issue by WillEhrendreich** [GustavEikaas/easy-dotnet.nvim#180]
This plugin is obviously incredible. great job. so so cool. 

The way the keymaps are handled circumvents the usual way keymaps are usually handled by most plugins.Typically, the plugin author exposes Commands that people can feel free to map to if they so choose. that makes them discoverable and usuable in other ways than you might have already thought of, and helps people create a more custom way these things are interacted with in different scenarios, like having them load only on attaching to certain buffer types, or under certain other circumstances like if something is building correctly or.. whatever. 

The way keymaps are handled here (pointing at local functions within keymaps.lua  
https://github.com/GustavEikaas/easy-dotnet.nvim/blob/11d48cce707756da73a4202f4985eefa6ad1f2c2/lua/easy-dotnet/test-runner/keymaps.lua#L289
) means that they aren't easily used outside of that local file's context. 
you pass function that's assigned to the lhs of the require("easy-dotnet.test-runner.render").options.mappings but give no way to reference what that function does outside of that exact keymapping, so if I don't want to set those on config, I have to copypaste that function into some other place I do my keymapping, which is techinically doable, but not a great api for users. This may mean also that they are assigned to the keymap in an order that's unexpected by people, either reassigning that keymap to something they already have, or by being reassigned when they didn't expect those keymaps to be reassigned because they thought it was taken care of by another place in code, etc. 

Would you mind making the mappings just call commands or public functions that are exposed publicly from your code? that would be amazing. 

great work on all this, again, I'm thrilled about how incredible this all looks.

**Comment by GustavEikaas** [comment]
For a lot of the keymaps this does make sense but for a few it kinda doesnt.  
exposing `Dotnet testrunner refresh` is no problem and it would make sense but for other keymaps like e.g `keymap.go_to_file.lhs` it doesnt? This keymap can only be invoked inside the testrunner buffer, I can technically make this a user command but im not sure I understand your usecase? Would you want to invoke the function for go_to_file not through a keymap?

Could you give me a specific example of something you want to do but cant?

**Comment by WillEhrendreich** [comment]
hmm.. well.. maybe in that instance go to file could be come the other way around, like, go_to_test_in_runner? then that could be a quick way to hop back and forth.. 

other than that, I'm pretty sure I had it take over some of my maps that were there. even after the test runner was closed.. though I can't seem to reproduce that right now..I'm not sure, sorry! lol. but certainly some of them I think do make sense to be public, if not all of them.

**Comment by GustavEikaas** [comment]
Ill look into it, it makes sense to make commands for a lot of them atleast. Thanks for the suggestion!

If you manage to reproduce the testrunner affecting your keybinds outside the testrunner buffer, please let me know because it shouldnt
```lua
vim.keymap.set('n', key, function()
      local line_num = vim.api.nvim_win_get_cursor(0)[1]
      local node = translateIndex(line_num, M.tree)
      if not node then
        error("Current line is not a node")
      end
      value(node, M)
    end, { 
    buffer = M.buf, --Buffer specific mapping
    noremap = true, 
    silent = true })

```

**Comment by GustavEikaas** [comment]
Just to clarify, all the keybindings for the testrunner are 100% configurable. So your use case is for using them outside the buffer?

**Comment by GustavEikaas** [comment]
So I wanted to do something about this. As I see it I want my exposing commands be in sync with my exposing lua functions at all times. I made a structure that generates the same exposing functions as commands in this PR #191. it pretty much means that I have manually verified every single command and lua function previously exposed. Hopefully this will make it easier in the future to have everything in sync. 

I would greatly appreciate if anyone could test out this branch and check for breaking changes before I merge it.🚀
